### PR TITLE
Add support for Android Blueprint

### PR DESCRIPTION
**Pull request by ambroisie**
I couldn't run the `update-lockfile.sh` script due to it trying to write to my read-only dotfiles, but I believe this should be correct.

### CODE REVIEW

File: queries/bp/injections.scm [hunk 1]
```diff
@@ -0,0 +1,4 @@
+((comment) @injection.content
+  (#set! injection.language "comment"))
+
+; vim: sw=2 foldmethod=marker
```
Comment (clason) [inline queries/bp/injections.scm:4]:
```suggestion
```

**Comment by clason** [comment]
There's no point in replicating upstream tests. Tests here should only cover specifics to this repo (indents, or in case upstream queries differ significantly, e.g., by a different query order ).

File: queries/bp/folds.scm [hunk 2]
```diff
@@ -0,0 +1,8 @@
+[
+  (list_expression)
+  (map_expression)
+  (module)
+  (select_expression)
+] @fold
+
+; vim: sw=2 foldmethod=marker
```
Comment (ObserverOfTime) [inline queries/bp/folds.scm:8]:
```suggestion
```

File: queries/bp/indents.scm [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+(module) @indent.begin
+
+(module
+  ")" @indent.branch)
+
+(module
+  "}" @indent.branch)
+
+; }}}
+; vim: sw=2 foldmethod=marker
```
Comment (ObserverOfTime) [inline queries/bp/indents.scm:44]:
```suggestion
```

File: queries/bp/highlights.scm [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+; Expressions {{{
+(map_expression
+  (property
+    field: (identifier) @property))
+
+(select_expression
+  "select" @keyword.conditional)
+
+; }}}
+; vim: sw=2 foldmethod=marker
```
Comment (ObserverOfTime) [inline queries/bp/highlights.scm:69]:
```suggestion
```

File: queries/bp/locals.scm [hunk 5]
```diff
@@ -0,0 +1,14 @@
...
+(map_expression
+  (property
+    field: (identifier) @local.definition.field))
+
+(assignment
+  left: (identifier) @local.definition.var)
+
+(identifier) @local.reference
+
+; vim: sw=2 foldmethod=marker
```
Comment (ObserverOfTime) [inline queries/bp/locals.scm:14]:
```suggestion
```

File: tests/query/highlights/bp/builtins.bp
Comment (ObserverOfTime) [inline tests/query/highlights/bp/builtins.bp:1]: You don't need to add highlight tests here if they're in the parser repo and the queries are the same.

File: queries/bp/highlights.scm [hunk 6]
```diff
@@ -0,0 +1,69 @@
+(comment) @comment
```
Comment (ObserverOfTime) [inline queries/bp/highlights.scm:1]:
```suggestion
(comment) @comment @spell
```

**Review by ObserverOfTime — APPROVED** [review]

**Comment by clason** [comment]
Indent tests are failing.

File: tests/indent/bp/select.bp [hunk 7]
```diff
@@ -0,0 +1,60 @@
...
+        default: 0,
+    }
+)
+
+foo = select((
+    arch(),
+    os(),
+), {
+    (default, default): [],
+})
```
Comment (lucario387) [inline tests/indent/bp/select.bp:35]: Can you drop this test for the time being? This has been a recurring issue for most languages that have this style of code.

Comment (ambroisie) [reply tests/indent/bp/select.bp:35]: What's wrong with keeping it as an expected failure?

Comment (clason) [reply tests/indent/bp/select.bp:35]: Noise, and CI load.

Comment (clason) [reply tests/indent/bp/select.bp:35]: This is _not_ resolved; we don't want to add new expected failures.

Comment (ambroisie) [reply tests/indent/bp/select.bp:35]: But I did remove the XFAIL?

File: tests/indent/bp_spec.lua [hunk 8]
```diff
@@ -0,0 +1,93 @@
...
+  softtabstop = 0,
+  expandtab = true,
+})
+
+describe("indent Blueprint:", function()
+  describe("whole file:", function()
+    runner:whole_file(".", {
+      expected_failures = {
+        "./select.bp",
+      },
```
Comment (clason) [inline tests/indent/bp_spec.lua:15]: I'm talking about this.

**Comment by lucario387** [comment]
Test failure aren't related.

**Review by lucario387 — APPROVED** [review]

**Pull request merged**